samples/camera/camerax/src/main/java/com/example/platform/camerax/basic/CameraXBasic.kt

```diff
@@ -0,0 +1,308 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.Button
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.isSpecified
+import androidx.compose.ui.geometry.takeOrElse
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.round
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.compose.LocalLifecycleOwner
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import coil.compose.rememberAsyncImagePainter
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.PermissionState
+import com.google.accompanist.permissions.PermissionStatus
+import com.google.accompanist.permissions.rememberPermissionState
+import com.google.accompanist.permissions.shouldShowRationale
+import kotlinx.coroutines.delay
+import java.util.UUID.randomUUID
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+/**
+ * This is a basic CameraX sample demonstrating how to use CameraX with Compose. It handles camera
+ * permissions, displays a camera preview, and captures photos.
+ */
+@OptIn(ExperimentalPermissionsApi::class)
+@Composable
+fun CameraXBasic(modifier: Modifier = Modifier) {
+    var showCapturedImage by remember { mutableStateOf<Uri?>(null) }
+    val cameraPermissionState = rememberPermissionState(Manifest.permission.CAMERA)
+    val imageCaptureCallbackExecutor: ExecutorService = remember { Executors.newSingleThreadExecutor() }
+    val viewModel = remember { CameraXBasicViewModel() }
+
+    DisposableEffect(Unit) {
+        onDispose {
+            imageCaptureCallbackExecutor.shutdown()
+        }
+    }
+
+    Box(modifier = Modifier.fillMaxSize()) {
```

Comment:
Build on the provided `modifier` param i.e. `Box(modifier = modifier.fillMaxSize()) {`